lib/api/include/irods/simpleQuery.h

```diff
@@ -25,6 +25,7 @@ typedef struct {
 #ifdef __cplusplus
 extern "C"
 #endif
-int rcSimpleQuery( rcComm_t *conn, simpleQueryInp_t *simpleQueryInp, simpleQueryOut_t **simpleQueryOut );
+__attribute__((deprecated("SimpleQuery is deprecated. Use GenQuery or SpecificQuery instead.")))
+int rcSimpleQuery(rcComm_t* conn, simpleQueryInp_t* simpleQueryInp, simpleQueryOut_t** simpleQueryOut);
```

Comment:
I'm not sure of the order of `extern "C"` and attributes.

Consider using the braced form of `extern "C"` to make things clear. For example:
```cpp
#ifdef __cplusplus
extern "C" {
#endif

__attribute(...)
int rcFunc();

#ifdef __cplusplus
} // extern "C"
#endif
```

I guess one way to verify the `extern "C"` is honored is to use `readelf` or `objdump` to see if the function name is mangled. If it's not mangled, then we're good.

Reply:
Seems okay to me
```
$ objdump -t ~/hdd/builds/irods/main/ubuntu-18.04/irods/lib/libirods_client.so | grep rcSimpleQuery
0000000000000000 l    df *ABS*  0000000000000000              rcSimpleQuery.cpp
000000000016f710 g     F .text  000000000000003e              rcSimpleQuery
$ readelf -s ~/hdd/builds/irods/main/ubuntu-20.04/irods/lib/libirods_client.so | grep rcSimpleQuery
  5899: 00000000001700e0    62 FUNC    GLOBAL DEFAULT   11 rcSimpleQuery
  2631: 0000000000000000     0 FILE    LOCAL  DEFAULT  ABS rcSimpleQuery.cpp
 10977: 00000000001700e0    62 FUNC    GLOBAL DEFAULT   11 rcSimpleQuery
```

Reply:
Good enough for me.